apps/prairielearn/src/sync/performance.ts

```diff
@@ -1,57 +1,31 @@
-const scopedData = {};
+const scopedData: Record<string, Record<string, number>> = {};
 
-/**
- * @param {string} scopeName
- */
-export function makePerformance(scopeName) {
+interface Performance {
+  start: (name: string) => void;
+  end: (name: string) => void;
+  timedAsync: <T>(name: string, asyncFunc: () => Promise<T>) => Promise<T>;
+}
+
+export function makePerformance(scopeName: string): Performance {
   if (!scopedData[scopeName]) {
     scopedData[scopeName] = {};
   }
 
   const scope = scopedData[scopeName];
 
-  /**
-   * @param {string} name
-   */
   function start(name: string): void {
-    scope[name] = new Date();
+    scope[name] = Date.now();
```

Comment:
While JS allows subtracting a date from a number, typecheck requires this subtraction to be between two numbers. Since `Date.now()` gives the time in milliseconds anyway, this conversion makes sense in this context and produces the same result.